### ISSUE TITLE
Feature request: Enable "Sticky" rows

### ISSUE THREAD

**Issue by answerquest** [koalyptus/HTML-Table-Filter-Generator#7]
Hi Max, Thanks for being so responsive so far! This is a request from a visitor to the budget analysis page where I'm using tablefilter : Once they find certain rows of interest, they want to "hold" that while clearing the search filter and looking for something else, "hold" another one, move to the next, etc. In this way from a table of say 1000 rows we want to manually pick out 15 rows that otherwise don't have a common search term / key / value etc to unite them programmatically.
So here's how the interface could be: checkboxes appear on the left edge of each row. If it gets checked, then that row stays in the view even if one clears the filter.

Alternatively, it could be a "star" feature, and another "favorites" table, identical in structure to the one we're working on, appears below that lists all the starred items. or re-appears in a special "selected" section above or below.

**Comment by koalyptus** [comment]
Hi there,
You should be able with the current API to implement your feature. Just as a proof of concept:
1. Go to http://tablefilter.free.fr/datademo.htm
2. Open the developers console 
4. In the javascript console type those commands:

<pre>
tf.hasVisibleRows = true;
tf.visibleRows = [5, 10];
// tf is the TF instance
</pre>
1. Perform different column searches
2. Expect the corresponding `sticky` rows to be always visible
3. Disable the sticky rows: <code>tf.hasVisibleRows = false;</code>
4. Perfom searches and expect to no longer see 'sticky' rows

Basically you should have the building blocks to implement this new feature and make customers happy :)

**Comment by koalyptus** [comment]
BTW, this script is now superseded by the <a href="https://github.com/koalyptus/TableFilter">TableFilter</a> code.

**Comment by answerquest** [comment]
Thanks. This works when there's no paging happening.. wonder if it could be made to work even when we do have paging? That had been a crucial element; don't want to give it up now.

I don't know how to implement this as a checkbox or link at every column. But maybe I can pre-render a link into the HTML before passing to tablefilter, so that clicking on it, that row is added to the tf.visibleRows array.

Still, please consider developing this as a feature. If you already have the `.visibleRows` parameter, then having tablefilter put checkboxes or similar toggles at every row is an obvious logical progression to make that ability available to the user. And it needs to work with paging.

Not sure exactly what's modernised in the new TableFilter repo.. I learned much more from the tablefilter.free.fr site as you've explained things so nicely there.

**Comment by koalyptus** [comment]
Unfortunately it is a known limitation (even documented :) ): the 'sticky' rows (`rows_always_visible` option) only work when paging is disabled. I will note this in the TODO list of `TableFilter` widget, as from now on I will no longer maintain this code. `TableFilter` has the same features as `HTML Table Filter Generator`. But the code base is now completely re-organised in a modular manner (ES6 modules), it enforces best practices (lint, tests etc.) and takes advantage of a variety of tools that make it easier to change, maintain and push to production (Grunt/Node, Webpack, Babel etc.). That's what I mean by 'modernised' :) It is just a starting point, the tech debt is still massive but at least there are some foundations!
My advice is to switch to TableFilter and as there is almost a parity in the options and it has already many [demos](https://github.com/koalyptus/TableFilter/tree/master/demos)

Concerning your feature, the use case seems limited for the moment, I would keep it as a 'mash-up' instead of a built-in feature. It definitely deserves a demo page when paging issue fixed.

**Comment by koalyptus** [comment]
If you are still interested, the 'sticky rows' fix when pagination enabled is now available under the <a href="https://github.com/koalyptus/TableFilter">TableFilter</a> repository.
